src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Windows.cs

```diff
@@ -24,6 +24,18 @@ public partial class MemoryMappedFile
             SafeFileHandle? fileHandle = fileStream != null ? fileStream.SafeFileHandle : null;
             Interop.Kernel32.SECURITY_ATTRIBUTES secAttrs = GetSecAttrs(inheritability);
 
+            if (fileStream != null)
```

Comment:
The original issue was reported for Unix special files. Why are these exceptions being added to the Windows file too?